config/main.py

```diff
@@ -167,11 +167,42 @@ def validate_namespace(namespace):
     else:
         return False
 
-def interface_alias_to_name(interface_alias):
+# Return the namespace where an interface belongs
+def get_intf_namespace(port):
+    """If it is a non multi-asic device, or if the interface is management interface
+       return '' ( DEFAULT_NAMESPACE ) which maps to current namespace ( in case of config commands
+       it is linux host )
+    """
+    if sonic_device_util.is_multi_npu() == False or port == 'eth0':
+        return DEFAULT_NAMESPACE
+
+    # If it is PortChannel or Vlan interface or Loopback, user needs to input the namespace.
+    if port.startswith("PortChannel") or port.startswith("Vlan") or port.startswith("Loopback"):
+        return None
+
+    if port.startswith("Ethernet"):
```

Comment:
get the port table, if it is in the port table then return the namespace, otherwise return non.

Reply:
Added a new API get_port_table_name() to get the table to be checked for the existence of the interface